source/text/Format.cpp

```diff
@@ -377,6 +377,64 @@ string Format::PlayTime(double timeVal)
 
 
 
+// Convert an ammo count into a short string for use in the ammo display.
+// Only the absolute value of a negative number is considered.
+string Format::AmmoCount(int64_t value)
+{
+	if(fabs(value) > SCIENTIFIC_THRESHOLD)
+	{
+		ostringstream out;
+		out.precision(2);
+		out << static_cast<double>(value);
+		return out.str();
+	}
+
+	int64_t absolute = abs(value);
+
+	if(absolute < 10000)
+		return to_string(value);
+
+	string result;
+	result.reserve(5);
+
+	// Handle numbers bigger than a million.
+	static const vector<char> SUFFIX = {'T', 'B', 'M', 'k'};
+	static const vector<int64_t> THRESHOLD = {1000000000000ll, 1000000000ll, 1000000ll, 1000ll};
+	for(size_t i = 0; i < SUFFIX.size(); ++i)
+		if(absolute >= THRESHOLD[i])
+		{
+			int head = absolute / THRESHOLD[i];
+			int tail = absolute % THRESHOLD[i];
```

Comment:
I think this needs to be 64-bit to make sure there's no value overflow. ~~Why are there no tests for tera?~~
```suggestion
			int64_t tail = absolute % THRESHOLD[i];
```